docs/Routing.md

```diff
@@ -125,74 +125,103 @@ export default App;
 
-But you may want to use another routing strategy, e.g. to allow server-side rendering. React-router offers various Router components to implement such routing strategies. If you want to use a different router, simply wrap it around your app. React-admin will detect that it's already inside a router, and skip its own router. 
+But you may want to use another routing strategy, e.g. to allow server-side rendering of individual pages. React-router offers various Router components to implement such routing strategies. If you want to use a different router, simply put your app in a create router function. React-admin will detect that it's already inside a router, and skip its own router. 
 
-```jsx
-import { BrowserRouter } from 'react-router-dom';
+```tsx
+import { RouterProvider, createBrowserRouter } from 'react-router-dom';
 import { Admin, Resource } from 'react-admin';
-
-const App = () => (
-    <BrowserRouter>
-        <Admin dataProvider={...}>
-            <Resource name="posts" />
-        </Admin>
-    </BrowserRouter>
-);
+import { dataProvider } from './dataProvider';
+
+const App = () => {
+    const router = createBrowserRouter([
+        {
+            path: "*",
+            element: (
+                <Admin dataProvider={dataProvider}>
+                    <Resource name="posts" />
+                </Admin>
+            ),
+        },
+    ]);
+    return <RouterProvider router={router} />;
+};
 ```
 
 ## Using React-Admin In A Sub Path
 
 React-admin links are absolute (e.g. `/posts/123/show`). If you serve your admin from a sub path (e.g. `/admin`), react-admin works seamlessly as it only appends a hash (URLs will look like `/admin#/posts/123/show`).
 
-However, if you serve your admin from a sub path AND use another Router (like `BrowserRouter` for instance), you need to set the `<BrowserRouter basename>` prop, so that react-admin routes include the basename in all links (e.g. `/admin/posts/123/show`).
+However, if you serve your admin from a sub path AND use another Router (like [`createBrowserRouter`](https://reactrouter.com/en/main/routers/create-browser-router) for instance), you need to set the `<Admin basename>` prop and the [`opts.basename`](https://reactrouter.com/en/main/routers/create-browser-router#optsbasename) of `createBrowserRouter` function, so that react-admin routes include the basename in all links (e.g. `/admin/posts/123/show`).
 
-```jsx
+```tsx
 import { Admin, Resource } from 'react-admin';
-
-const App = () => (
-    <BrowserRouter basename="/admin">
-        <Admin dataProvider={...}>
-            <Resource name="posts" />
-        </Admin>
-    </BrowserRouter>
-);
+import { RouterProvider, createBrowserRouter } from 'react-router-dom';
+import { dataProvider } from './dataProvider';
+
+const App = () => {
+    const router = createBrowserRouter(
+        [
+            {
+                path: "*",
+                element: (
+                    <Admin basename="/admin" dataProvider={dataProvider}>
```

Comment:
Do we need to declare the basename in Admin if it's already declared in the router?

Reply:
Yes we do.
`<Admin basename>` is used to do both:
- Set the basename in the default router (being createHashRouter)
- Add the basename as prefix to all redirections happening in react-admin (e.g. clicking on a CreateButton)

When we provide our own router, we need to set the basename to our own router, but the `<Admin basename>` is still required to compute the redirections properly.

Reply:
Okay so I actually tested this case locally, and it turns out @fzaninotto was right! `<Admin basename>` must be left empty, otherwise the path is added twice (`/admin/admin/posts/1/show`).

```suggestion
                    <Admin dataProvider={dataProvider}>
```